src/lib/services/feature-toggle-service.ts

```diff
@@ -271,6 +277,14 @@ class FeatureToggleService {
         }
     }
 
+    async validateStrategyType(
+        strategyName: string | undefined,
+    ): Promise<void> {
+        if (strategyName !== undefined) {
+            await this.strategyStore.get(strategyName);
```

Comment:
is it depending on store throwing an error? 
I'd rather call exists method on store and explicitly throw and error BadDataError. Those DB errors get propagated to the UI which is not nice. It's better to wrap it into some HTTP friendly error

Reply:
Updated